app/vmui/packages/vmui/src/pages/CustomPanel/QueryConfigurator/QueryConfigurator.tsx

```diff
@@ -113,6 +123,51 @@ const QueryConfigurator: FC<QueryConfiguratorProps> = ({
     handleToggleHideQuery(e, i);
   };
 
+  const { serverUrl } = useAppState();
+
+  const handlePrettifyQuery = async (el: HTMLButtonElement, i: number) => {
+
+    const oldQuery = encodeURIComponent(stateQuery[i]);
+    let response: Response;
+    try {
+      response = await fetch(`${serverUrl}/prettify-query?query=${oldQuery}`);
+    } catch (e) {
+      const newQueryErrors = [...queryErrors];
+      newQueryErrors[i] = `${e}`;
+      setQueryErrors(newQueryErrors);
+      return;
+    }
+
+    if (response.status != 200) {
+      const newQueryErrors = [...queryErrors];
+      newQueryErrors[i] = "Error requesting /prettify-query, status: " + response.status;
+      setQueryErrors(newQueryErrors);
+    }
+
+    const data = await response.json();
+
+    if (data["status"] == "success") {
+      const newQueryErrors = [...queryErrors];
+      newQueryErrors[i] = "";
+      setQueryErrors(newQueryErrors);
+
+      const newStateQuery = [...stateQuery];
+      newStateQuery[i] = data["query"];
+      setStateQuery(newStateQuery);
+    } else {
+      const newQueryErrors = [...queryErrors];
+      newQueryErrors[i] = data["msg"];
+      setQueryErrors(newQueryErrors);
+    }
+
+  };
+
+  const createHandlerPrettifyQuery = async (i: number) => {
```

Comment:
`createHandlerPrettifyQuery` returns a `Promise`, which leads to a type error:

<img width="939" alt="image" src="https://github.com/VictoriaMetrics/VictoriaMetrics/assets/29711459/1eda6e83-e0b9-46d1-a98a-8a00c9563758">